.github/workflows/maven_deploy_template.yml

```diff
@@ -0,0 +1,121 @@
+        description: 'Additional arguments passed to maven'
+        required: false
+        type: string
+        default: ''
+      checkout-lfs:
+        description: 'Checkout files from Git LFS'
+        required: false
+        type: boolean
+        default: false
+      version:
+        description: 'Version used for maven CI friendly versions'
+        required: true
+        type: string
+    secrets:
+      MAVEN_GPG_PRIVATE_KEY:
+        required: true
+        description: 'Private GPG key to sign artifacts with'
+      MAVEN_GPG_PASSPHRASE:
+        required: true
+        description: 'Passphrase for private gpg key'
+      MAVEN_CENTRAL_USER_NAME:
+        required: true
+        description: 'User name for maven central for deployment'
+      MAVEN_CENTRAL_TOKEN:
+        required: true
+        description: 'Token for maven central deployment'
+
+env:
+  MAVEN_CLI_OPTS: '--batch-mode --errors --fail-at-end -T 1C'
+
+jobs:
+  Validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate tagged version
+        run: |
+          RE_SEMVER="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$"
+          
+          if [[ ! "${{ inputs.version }}" =~ $RE_SEMVER ]]; then
+            echo "ERROR tag with value ${{ inputs.version }} is an invalid release or semantic version"
+            echo "Release requires a current tag containing a valid SemVer 2.0.0 without pre-release version"
+            exit 1
+          fi
+      - name: Checkout project
+        uses: actions/checkout@v3
+        with:
+          lfs: ${{ inputs.checkout-lfs }}
+      - name: Set up JDK ${{ inputs.jdk-version }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ inputs.jdk-version }}
+          distribution: temurin
+          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+          server-username: MAVEN_USERNAME # env variable for username in deploy
+          server-password: MAVEN_CENTRAL_TOKEN # env variable for token in deploy
+      - name: Validate project version
+        run: |
+          PROJECT_VERSION=$(mvn -Dexpression=project.version -q -DforceStdout -Drevision=${{ inputs.version }} help:evaluate)
+          if [[ ! $PROJECT_VERSION == ${{ inputs.version }} ]]; then
+            echo "ERROR tag with value ${{ inputs.version }} does not correspond to project version $PROJECT_VERSION"
+            exit 1
+          fi
+      - name: Check release existence
+        env:
+          MAVEN_USERNAME: ${{ secrets.MAVEN_CENTRAL_USER_NAME }}
+          MAVEN_CENTRAL_TOKEN: ${{ secrets.MAVEN_CENTRAL_TOKEN }}
+        run: >
+          mvn $MAVEN_CLI_OPTS $MAVEN_OPTS
+          -Dexists.failIfExists=true
+          -Dexists.skipIfSnapshot=true
+          -Dexists.serverId=ossrh
+          -Dexists.snapshotServerId=ossrh
+          -Drevision=$GITHUB_REF_NAME
+          -Dcentraldeploy=true
+          org.honton.chas:exists-maven-plugin:0.7.0:remote  
```

Comment:
Hier wird dann das Staging Repo geprüft in dem keine Artefakte nach Release vorliegen? Eher gegen http://repo1.maven.org/maven2 prüfen. Bitte testen. Vermutlich wird das Plugin nicht benötigt da Maven Central doppelte Versionen sowieso nicht zulässt.

Reply:
Ich habe die URL angepasst. Das Plugin habe ich mit eingebaut, da so eine Pipeline frühzeitig abgebrochen kann, wenn ein Release bereits vorhanden ist und so nicht unnötigerweise die Artefakte (Jars, Docs, Sources, SBOMs und Signaturen) gebaut werden.